mlir/lib/IR/AsmPrinter.cpp

```diff
@@ -195,6 +196,10 @@ struct AsmPrinterOptions {
       "mlir-print-unique-ssa-ids", llvm::cl::init(false),
       llvm::cl::desc("Print unique SSA ID numbers for values, block arguments "
                      "and naming conflicts across all regions")};
+
+  llvm::cl::opt<bool> useNameLocAsPrefix{
+      "mlir-use-nameloc-as-prefix", llvm::cl::init(false),
+      llvm::cl::desc("Print SSA IDs using NameLocs as prefixes")};
```

Comment:
Is this all or only those not set via OpAsmInterface?

Reply:
It's all - this actually takes precedence over OpAsmInterface. I figured if you turn this on then you want even more legibility than OpAsmInterface provides.